src/main/java/de/tum/in/www1/artemis/web/rest/LtiResource.java

```diff
@@ -137,7 +141,7 @@ public void launch(@ModelAttribute LtiLaunchRequestDTO launchRequest, @PathVaria
         // If the current user was created within the last 15 minutes, we just created the user
         // Display a welcome message to the user
         boolean isNewUser = SecurityUtils.isAuthenticated()
-                && TimeUnit.SECONDS.toMinutes(ZonedDateTime.now().toEpochSecond() - userRepository.getUser().getCreatedDate().getEpochSecond()) < 15;
+                && TimeUnit.SECONDS.toMinutes(timeService.now().toEpochSecond() - userRepository.getUser().getCreatedDate().getEpochSecond()) < 15;
```

Comment:
The big advantage is that we can mock now. If you look into one test case, I set now 20 min into the future. This can be very handy. There are other ways to do that but those are a bit more difficult to implement and require more changes (mainly the use of a clock class)